### PR TITLE
Implemented OnBoarding Wizard

### PR DESCRIPTION
**Pull request by JWKaiqi**
Issue ga4gh/dockstore#743 and ga4gh/dockstore#810 has been updated.

- Updated the setup page, which now includes the on-boarding wizard.
- Also fixed the text overflow problem for the starred page, when the title in the starred tools and workflows page is too long, the panel truncate the title with "..."

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -70,6 +71,8 @@ import { StarringModule } from './starring/starring.module';
 import { StargazersModule } from './stargazers/stargazers.module';
 import { ListentryModule } from './listentry/listentry.module';
 import {AdvancedsearchComponent} from './search/advancedsearch/advancedsearch.component';
+import { DownloadcliclientComponent } from './loginComponents/onboarding/downloadcliclient/downloadcliclient.component';
+import { SetupcompleteComponent } from './loginComponents/onboarding/setupcomplete/setupcomplete.component';
```
Comment (garyluu) [inline src/app/app.module.ts]: Try to use upper camel case for component names.  (i.e. SetupCompleteComponent)

File: src/app/loginComponents/token.service.ts [hunk 2]
```diff
@@ -26,4 +26,49 @@ export class TokenService {
     return this.httpService.deleteAuth(deleteTokenUrl);
   }
 
+  getWebServiceVersion() {
+    const url = `${ Dockstore.API_URI }/api/ga4gh/v1/metadata`;
```
Comment (garyluu) [inline src/app/loginComponents/token.service.ts]: this already exists in versions.service.ts, merge the two

File: src/app/loginComponents/token.service.ts [hunk 3]
```diff
@@ -26,4 +26,49 @@ export class TokenService {
...
+    return this.httpService.getResponse(url);
+  }
+
+  getUserTokens(userId) {
+    const url = `${ Dockstore.API_URI }/users/${userId}/tokens`;
```
Comment (garyluu) [inline src/app/loginComponents/token.service.ts]: this is also available in user.service.ts

File: src/app/loginComponents/onboarding/onboarding.component.html [hunk 4]
```diff
@@ -1,4 +1,26 @@
...
+  <div *ngIf="curStep == 2">
+    <app-downloadcliclient></app-downloadcliclient>
+  </div>
+  <div *ngIf="curStep == 3">
+    <app-setupcomplete></app-setupcomplete>
+  </div>
+  <nav class="pull-right">
+    <ul class="pager">
+      <li *ngIf="curStep > 1">
+        <a style="cursor: pointer" (click)="prevStep()"><span class="glyphicon glyphicon-chevron-left"></span>Previous</a>
```
Comment (garyluu) [inline src/app/loginComponents/onboarding/onboarding.component.html]: avoid inline styles when possible

File: src/app/app.module.ts [hunk 5]
```diff
@@ -89,7 +91,9 @@ import {AdvancedsearchComponent} from './search/advancedsearch/advancedsearch.co
     AuthComponent,
     TokensComponent,
     StarredentriesComponent,
-    AdvancedsearchComponent
```
Comment (denis-yuen) [inline src/app/app.module.ts]: Hmmm,  `StarredEntriesComponent` and `AdvancedSearchComponent` for consistency?

File: src/app/loginComponents/onboarding/onboarding.component.html (hunk 4 above)
Comment (garyluu) [inline src/app/loginComponents/onboarding/onboarding.component.html]: On second thought...perhaps a button would make more sense.  That would include the pointer by default.  @denis-yuen , any reasons why this component in UI1 is not a button?

Comment (JWKaiqi) [inline src/app/loginComponents/onboarding/onboarding.component.html]: @garyluu need to modify the css if we use a button instead.

Comment (denis-yuen) [inline src/app/loginComponents/onboarding/onboarding.component.html]: @garyluu  No idea, try it out

**Comment by JWKaiqi** [comment]
Updated pull request:
- merged duplicated functions into one service.
- changed <a> to a <button> for the navigation buttons of the OnBording Wizard page (i.e: previous, next, and finish)
- changed component names to upper case (StarredEntriesComponent and AdvancedSearchComponent).

**Pull request merged**